src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -417,14 +417,26 @@ public void execute(StartupChecksOptions options) throws StartupException
         }
     };
 
-    public static final StartupCheck initSigarLibrary = new StartupCheck()
+    public static final StartupCheck checkProcessEnvironment = new StartupCheck()
     {
         @Override
         public void execute(StartupChecksOptions options)
         {
             if (options.isDisabled(getStartupCheckType()))
                 return;
-            SigarLibrary.instance.warnIfRunningInDegradedMode();
+
+            Optional<String> degradations = new SystemInfo().isDegraded();
+
+            if (degradations.isPresent())
+            {
+                String message = "Cassandra server running in degraded mode. " + degradations.get();
+                if (CassandraRelevantProperties.TEST_IGNORE_PROCESS_ENVIRONMENT_CHECK.getBoolean())
+                    logger.warn(message);
+                else
+                    logger.info(message);
+            }
+            else
+                logger.info("Checked OS settings and found them configured for optimal performance.");
```

Comment:
nit: why?  we didn't need this before.

Reply:
@michaelsembwever I see same log here

https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/utils/SigarLibrary.java#L179

If we are all good, we just print this, as it was before.

Reply:
Cool, thanks for pointing it out.

Reply:
I agree with you guys, I think we should just print warn if it is running in degraded mode and that's it. No info/warn distinction, no positive message when everything is ok.

Reply:
I consider this is just something which shouldn't be even a startup check because it does not stop the execution in certain cases. This is also a new Cassandra version so behaviour changes are ok. This adds extra complexity, extra log messages and I cannot really see any value in that. I think we should limit the log messages visible for the user rather than extending them and printing obvious information

Reply:
@jacek-lewandowski 

I think that make this a startup check is just fine for now. Let's just rewrite the guts of it for now ... There is as lot of startup checks which are not stopping node from starting. For example `checkReadAheadKbSetting` just (at most) logs warnings ... so based on that logic, that one does not qualify for being a startup check either. Similar to `checkLz4Native` etc ...

I think for now this is just fine. We could then probably consolidate these checks so they indeed stop the bootstrap if they fail and we could extract the rest to some other logic. I think that we are dealing primarily with "how things were".

Reply:
@Claudenw  @jacek-lewandowski 

To wrap it up, I think whole logic might be just replaced by this: 

   new SystemInfo().isDegraded()
                    .ifPresent(msgs -> logger.warn("Cassandra server running in degraded mode. " + msgs));

and we remove `TEST_IGNORE....` altogether.

Reply:
Why not? Then we need to remove all these checks as well and find new place for them with whole machinery around that.

checkJemalloc
checkLz4Native
checkJMXPorts
checkJMXProperties
inspectJvmOptions
checkMaxMapCount
checkReadAheadKbSetting

Why not to just do it as proposed here and then we can elaborate on that in a different ticket?

Keep in mind that we are rewriting the check, we are not implementing a new one. This will save us couple MBs from the tarball too.

I understand your point but extracting this from startup checks is not the objective here at the moment ...

Reply:
> Why not to just do it as proposed here and then we can elaborate on that in a different ticket?
>
> Keep in mind that we are rewriting the check, we are not implementing a new one. 

yes.  startup check for this ticket works for me.  let's not drag the ticket on forever.

Reply:
ok, this is not objective of this ticket, indeed

Reply:
yes, I'm -1 on keeping info; If we really need something like that, we should have a generic mechanism which would just say: "Running startup check <name>" for each check consistently (out of scope).

Reply:
I am for keeping it there until we found more robust way how to inform a user what startup check is being run. I can imagine all would log they are executed on DEBUG or similar, then this like can go away too.

Reply:
> This would be an change in behavior and we are all OK with that? I want to verify before removing- Personally, I think it should stay.

please keep existing behaviour (don't remove).
we never know who's using the logs for want, there needs to be good reason to change.